service/src/test/java/bio/terra/drshub/config/DrsProviderTest.java

```diff
@@ -0,0 +1,82 @@
+package bio.terra.drshub.config;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.drshub.BaseTest;
+import bio.terra.drshub.models.AccessMethodConfigTypeEnum;
+import bio.terra.drshub.models.AccessUrlAuthEnum;
+import bio.terra.drshub.models.BondProviderEnum;
+import io.github.ga4gh.drs.model.AccessMethod.TypeEnum;
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+
+public class DrsProviderTest extends BaseTest {
+
+  private final ProviderAccessMethodConfig testAccessMethodConfig =
+      ProviderAccessMethodConfig.create()
+          .setType(AccessMethodConfigTypeEnum.gs)
+          .setAuth(AccessUrlAuthEnum.fence_token)
+          .setFetchAccessUrl(true);
+
+  @Test
+  void testShouldFetchFenceAccessTokenReturnsTrue() {
+    // returns true when there are no problems identifying the relevant configs
+    var accessMethods = new ArrayList<ProviderAccessMethodConfig>();
+    accessMethods.add(testAccessMethodConfig);
```

Comment:
Can you use `List.of` for these?

Reply:
I agree that would be tidier. I did try that and the issue is that `List.of` method outputs a `List` but our [DrsProviderInterface config field](https://github.com/DataBiosphere/terra-drs-hub/blob/1f454b6afb1dd4bd311d4a8151083deed3886b5d/service/src/main/java/bio/terra/drshub/config/DrsProviderInterface.java#L27) expects an `ArrayList`. If you change the `DrsProviderInterface` to expect a List instead, it complains that it fails to "load the ApplicationContext" because it "cannot bind the property" because there's more than one implementation for the List interface. 

If you have any thoughts on how to get around that, I'd be happy to look into them. Otherwise, I might just leave this as is.

Reply:
I found this works:
```java
var accessMethods = new ArrayList<>(List.of(testAccessMethodConfig));
```

Also, could we declare `accessMethods` once as `final`?

Reply:
Yeah, that sounds like a good plan! Thanks for the suggestion!

Reply:
Let me know if the changes I made are what you had in mind.